modules/siddhi-extensions/input-transports/jms-input-transport/src/main/java/org/wso2/siddhi/extension/input/transport/jms/JMSInputTransport.java

```diff
@@ -0,0 +1,217 @@
+                optionHolder.getOrCreateOption(
+                        JMSInputTransportConstants.ADAPTER_JMS_SUBSCRIPTION_DURABLE, "").getValue());
+        transportProperties.put(JMSInputTransportConstants.ADAPTER_JMS_DURABLE_SUBSCRIBER_CLIENT_ID,
+                optionHolder.getOrCreateOption(
+                        JMSInputTransportConstants.ADAPTER_JMS_DURABLE_SUBSCRIBER_CLIENT_ID, "").getValue());
+        if (jmsProperties != null) {
+            transportProperties.putAll(jmsProperties);
+            if (optionHolder.isOptionExists(JMSInputTransportConstants.ADAPTER_JMS_CONCURRENT_CONSUMERS)) {
+                try {
+                    minThreadPoolSize = Integer.parseInt(optionHolder.validateAndGetStaticValue(
+                            JMSInputTransportConstants.ADAPTER_JMS_CONCURRENT_CONSUMERS));
+                } catch (NumberFormatException e) {
+                    log.error("Invalid JMS Property: " + JMSInputTransportConstants.ADAPTER_JMS_CONCURRENT_CONSUMERS +
+                            ", ignoring configuration and using default for JMS output event adaptor...");
+                }
+            }
+            if (optionHolder.isOptionExists(JMSInputTransportConstants.ADAPTER_JMS_MAX_CONCURRENT_CONSUMERS)) {
+                try {
+                    maxThreadPoolSize = Integer.parseInt(optionHolder.validateAndGetStaticValue(
+                            JMSInputTransportConstants.ADAPTER_JMS_MAX_CONCURRENT_CONSUMERS));
+                } catch (NumberFormatException e) {
+                    log.error("Invalid JMS Property: " + JMSInputTransportConstants.ADAPTER_JMS_MAX_CONCURRENT_CONSUMERS
+                            + ", ignoring configuration and using default for JMS output event adaptor...");
+                }
+            }
+        }
+        if (secureJmsProperties != null) {
+            transportProperties.putAll(secureJmsProperties);
+        }
+        return transportProperties;
+    }
+
+    private Map<String, String> extractProperties(String properties) {
+        if (properties == null || properties.trim().length() == 0) {
+            return null;
+        }
+
+        String[] entries = properties.split(JMSInputTransportConstants.PROPERTY_SEPARATOR);
+        String[] keyValue;
+        Map<String, String> result = new HashMap<String, String>();
+        for (String property : entries) {
+            try {
+                keyValue = property.split(JMSInputTransportConstants.ENTRY_SEPARATOR, 2);
+                result.put(keyValue[0].trim(), keyValue[1].trim());
+            } catch (Exception e) {
+                log.warn("JMS property '" + property + "' is not defined in the correct format.", e);
+            }
+        }
+        return result;
+    }
+
+    private Hashtable<String, String> convertMapToHashTable(Map<String, String> map) {
+        Hashtable<String, String> table = new Hashtable<>();
+        for (Map.Entry<String, String> stringStringEntry : map.entrySet()) {
```

Comment:
isn't it too complex to use for each here?
Since there's an if condition as well?